src/dex/aave-v3/utils.ts

```diff
@@ -0,0 +1,153 @@
+import { BytesLike } from 'ethers';
+import { defaultAbiCoder, Interface } from '@ethersproject/abi';
+import {
+  extractSuccessAndValue,
+  stringDecode,
+  uint8ToNumber,
+} from '../../lib/decoders';
+import {
+  MultiCallParams,
+  MultiResult,
+  MultiWrapper,
+} from '../../lib/multi-wrapper';
+import Web3 from 'web3';
+import POOL_ABI from '../../abi/AaveV3_lending_pool.json';
+import { AaveToken } from './types';
+
+export const decodeATokenFromReserveData = (
+  result: MultiResult<BytesLike> | BytesLike,
+): string => {
+  const [isSuccess, toDecode] = extractSuccessAndValue(result);
+
+  if (!isSuccess || toDecode === '0x') {
+    throw new Error(
+      'Could not extract value from struct DataTypes.ReserveData.',
+    );
+  }
+
+  const decoded = defaultAbiCoder.decode(
+    [
+      'tuple(uint256) configuration',
```

Comment:
better to use common [ABI file](https://github.com/paraswap/paraswap-dex-lib/blob/478d6b231bc55e569adea4cc7baf7a50d8cc305b/src/abi/AaveV3_lending_pool.json#L994) and decode via [Interface](https://docs.ethers.org/v5/api/utils/abi/interface/)

Reply:
I am not sure it is possible to pass custom `Interface` to decoder function. Or it would be something predefined in Utils with closure and will be looking ugly.
In other places we always use this approach of decoding with custom types :)